File: src/main/resources/statics/js/modules/small/smallpurorderdetail.js
$(function () {
    $("#jqGrid").jqGrid({
        url: baseURL + 'small/smallpurorderdetail/list',
        datatype: "json",
        colModel: [			
			{ label: 'id', name: 'id', index: 'id', width: 50, key: true },
			{ label: '商品(skuid)', name: 'skuId', index: 'sku_id', width: 80 }, 			
			{ label: '商品（spuid）', name: 'spuId', index: 'spu_id', width: 80 }, 			
			{ label: '订单id', name: 'purorderId', index: 'purorder_id', width: 80 }, 			
			{ label: '订单编号', name: 'purorderNo', index: 'purorder_no', width: 80 }, 			
			{ label: '商品spu名称', name: 'spuTitle', index: 'spu_title', width: 80 }, 			
			{ label: '商品sku名称', name: 'skuTitle', index: 'sku_title', width: 80 }, 			
			{ label: '条码', name: 'barCode', index: 'bar_code', width: 80 }, 			
			{ label: '计量单位', name: 'unit', index: 'unit', width: 80 }, 			
			{ label: '数量', name: 'num', index: 'num', width: 80 }, 			
			{ label: '原价', name: 'originalPrice', index: 'original_price', width: 80 }, 			
			{ label: '现价', name: 'price', index: 'price', width: 80 }, 			
			{ label: 'spu图片', name: 'spuImg', index: 'spu_img', width: 80 }, 			
			{ label: '创建时间', name: 'createTime', index: 'create_time', width: 80 }, 			
			{ label: '修改时间', name: 'modifyTime', index: 'modify_time', width: 80 }			
        ],
		viewrecords: true,
        height: 385,
        rowNum: 10,
		rowList : [10,30,50],
        rownumbers: true, 
        rownumWidth: 25, 
        autowidth:true,
        multiselect: true,
        pager: "#jqGridPager",
        jsonReader : {
            root: "page.list",
            page: "page.currPage",
            total: "page.totalPage",
            records: "page.totalCount"
        },
        prmNames : {
            page:"page", 
            rows:"limit", 
            order: "order"
        },
        gridComplete:function(){
        	//隐藏grid底部滚动条
        	$("#jqGrid").closest(".ui-jqgrid-bdiv").css({ "overflow-x" : "hidden" }); 
        }
    });
});

var vm = new Vue({
	el:'#icloudapp',
	data:{
		showList: true,
		title: null,
		smallPurorderDetail: {}
	},
	methods: {
		query: function () {
			vm.reload();
		},
		add: function(){
			vm.showList = false;
			vm.title = "新增";
			vm.smallPurorderDetail = {};
		},
		update: function (event) {
			var id = getSelectedRow();
			if(id == null){
				return ;
			}
			vm.showList = false;
            vm.title = "修改";
            
            vm.getInfo(id)
		},
		saveOrUpdate: function (event) {
		    $('#btnSaveOrUpdate').button('loading').delay(1000).queue(function() {
                var url = vm.smallPurorderDetail.id == null ? "small/smallpurorderdetail/save" : "small/smallpurorderdetail/update";
                $.ajax({
                    type: "POST",
                    url: baseURL + url,
                    contentType: "application/json",
                    data: JSON.stringify(vm.smallPurorderDetail),
                    success: function(r){
                        if(r.code === 0){
                             layer.msg("操作成功", {icon: 1});
                             vm.reload();
                             $('#btnSaveOrUpdate').button('reset');
                             $('#btnSaveOrUpdate').dequeue();
                        }else{
                            layer.alert(r.msg);
                            $('#btnSaveOrUpdate').button('reset');
                            $('#btnSaveOrUpdate').dequeue();
                        }
                    }
                });
			});
		},
		del: function (event) {
			var ids = getSelectedRows();
			if(ids == null){
				return ;
			}
			var lock = false;
            layer.confirm('确定要删除选中的记录？', {
                btn: ['确定','取消'] //按钮
            }, function(){
               if(!lock) {
                    lock = true;
		            $.ajax({
                        type: "POST",
                        url: baseURL + "small/smallpurorderdetail/delete",
                        contentType: "application/json",
                        data: JSON.stringify(ids),
                        success: function(r){
                            if(r.code == 0){
                                layer.msg("操作成功", {icon: 1});
                                $("#jqGrid").trigger("reloadGrid");
                            }else{
                                layer.alert(r.msg);
                            }
                        }
				    });
			    }
             }, function(){
             });
		},
		getInfo: function(id){
			$.get(baseURL + "small/smallpurorderdetail/info/"+id, function(r){
                vm.smallPurorderDetail = r.smallPurorderDetail;
            });
		},
		reload: function (event) {
			vm.showList = true;
			var page = $("#jqGrid").jqGrid('getGridParam','page');
			$("#jqGrid").jqGrid('setGridParam',{ 
                page:page
            }).trigger("reloadGrid");
		}
	}
});

File: src/main/java/com/icloud/modules/small/entity/SmallPurorderDetail.java
package com.icloud.modules.small.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;

/**
 * 采购单明细
 * 
 * @author zdh
 * @email yyyyyy@cm.com
 * @date 2020-10-06 20:50:58
 */
@Data
@TableName("t_small_purorder_detail")
public class SmallPurorderDetail implements Serializable {
	private static final long serialVersionUID = 1L;

   	   /*  */
       @TableId(value="id", type= IdType.AUTO)
       private Long id;
   	   	   /* 商品(skuid) */
       @TableField("sku_id")
       private Long skuId;
   	   	   /* 商品（spuid） */
       @TableField("spu_id")
       private Long spuId;
   	   	   /* 订单id */
       @TableField("purorder_id")
       private Long purorderId;
   	   	   /* 订单编号 */
       @TableField("purorder_no")
       private String purorderNo;
   	   	   /* 商品spu名称 */
       @TableField("spu_title")
       private String spuTitle;
   	   	   /* 商品sku名称 */
       @TableField("sku_title")
       private String skuTitle;
   	   	   /* 条码 */
       @TableField("bar_code")
       private String barCode;
   	   	   /* 计量单位 */
       @TableField("unit")
       private String unit;
   	   	   /* 数量 */
       @TableField("num")
       private Integer num;
   	   	   /* 原价 */
       @TableField("original_price")
       private BigDecimal originalPrice;
   	   	   /* 现价 */
       @TableField("price")
       private BigDecimal price;
        /* 现价 */
       @TableField("shop_id")
       private Long shopId;

   	   	   /* spu图片 */
       @TableField("spu_img")
       private String spuImg;
   	   	   /* 创建时间 */
       @TableField("create_time")
       private Date createTime;
   	   	   /* 修改时间 */
       @TableField("modify_time")
       private Date modifyTime;
   	
}
